lib/source_file.h

```diff
@@ -126,6 +127,14 @@ class SourceInfo final {
     int line = -1;
     int column = -1;
     cstring srcBrief = ""_cs;
+
+    // add a comment to 'before' or 'after' array of comments
+    void addBefore(Util::Comment *comment) const;
+    void addAfter(Util::Comment *comment) const;
+
+    [[nodiscard]] const std::vector<Util::Comment *> &accessBefore() const;
+    [[nodiscard]] const std::vector<Util::Comment *> &accessAfter() const;
```

Comment:
I suggest naming this `getCommentsBefore`/`getCommentsAfter` it is clearer and I don't think there is "`access`" used elsewhere in the codebase.